### ISSUE TITLE
Remote Code Execution vulnerability

### ISSUE THREAD

**Issue by vvanmol** [locutusjs/locutus#420]
Hello, 

I haven't seen any open issue about this so I'm creating this ticket, but Snyk reported a vulnerability about this package.  
Please find more details about it here: https://snyk.io/vuln/SNYK-JS-LOCUTUS-575119

**Comment by kvz** [comment]
Is this the issue that needs fixing? https://github.com/locutusjs/locutus/issues/395

**Comment by vvanmol** [comment]
Seems to be related to the same function, but not the same issue.  Here it's about sanitizing two single quotes characters as explained in this Blog post: https://reallinkers.github.io/CVE-2020-13619/

The following code seems to exploit the vulnerability (taken from the node sample in the blog post above):

```javascript
var phpExec = require('locutus/php/exec');
var child = require('child_process');

var directory = "/home'; whoami;''"
var command = "ls -la " + phpExec.escapeshellarg(directory);
console.log(command);

child.exec(command, function (error,stdout,stderr) {
	console.log(stdout);
	console.log(stderr);
});
```

**Comment by divinity76** [comment]
here is how i would implement it personally, 

```js
function escapeshellarg(arg){
  if(arg.indexOf("\x00") !== -1) {
    throw new Error('escapeshellarg(): Argument #1 ($arg) must not contain any null bytes');
  }
  return "'"+arg.replace(/\'/g,'\'\\\'\'')+"'";
}
```
for one, it's much closer to how PHP's escapeshellarg() works, and second, it's not vulnerable to the above shell execution exploit, correctly generating the command
```
$ ls -la '/home'\''; whoami;'\'''\'''
ls: cannot access '/home'\''; whoami;'\'''\''': No such file or directory
```

**Comment by divinity76** [comment]
proposed a fix in #426

**Comment by divinity76** [comment]
uhm.. guys? can i get some feedback?

**Comment by vvanmol** [comment]
Hi @divinity76 

Your solution seems to be good, and you confirmed it fixes the vulnerability.  Not sure I can do anything from my side, but waiting for maintainers...

**Comment by kvz** [comment]
Thank you! Merged and released as locutus@2.0.13
